mbedtls/src/rng/rdrand.rs

```diff
@@ -7,33 +7,87 @@
  * according to those terms. */
 
 use core::slice::from_raw_parts_mut;
+
 use mbedtls_sys::types::raw_types::{c_int, c_uchar, c_void};
 use mbedtls_sys::types::size_t;
 
+// Intel documentation claims that if hardware is working RDRAND will produce output
+// after at most 10 attempts
+const RDRAND_READ_ATTEMPTS: usize = 10;
+
+// Intel does not document the number of times RDSEED might consecutively fail, but in
+// example code uses 75 as the upper bound.
+const RDSEED_READ_ATTEMPTS: usize = 75;
+
+fn call_cpu_rng<T, F>(attempts: usize, intrin: unsafe fn(&mut T) -> i32, cast_to_usize: F) -> Option<usize>
+where
+    T: Sized + Copy + Default,
+    F: FnOnce(T) -> usize,
+{
```

Comment:
Maybe you can add an assert equating the size of `T` and `usize` for good measure